src/main/java/uk/gov/companieshouse/items/orders/api/validator/RequestValidator.java

```diff
@@ -92,6 +90,27 @@
         return errors;
     }
 
+    /**
+     * Validates the collection delivery related fields on the options provided.
+     * @param options the options to be validated
+     * @return the resulting errors, which will be empty if the fields are found to be valid
+     */
+    List<String> getCollectionDeliveryValidationErrors(final CertificateItemOptions options) {
+        final List<String> errors = new ArrayList<>();
+        if (options.getDeliveryMethod() == COLLECTION) {
+            if (options.getCollectionLocation() == null) {
+                errors.add("collection_location: must not be null when delivery method is collection");
+            }
+            if (isBlank(options.getForename())) {
+                errors.add("forename: must not be blank when delivery method is collection");
+            }
+            if (isBlank(options.getSurname())) {
+                errors.add("surname: must not be blank when delivery method is collection");
+            }
+        }
```

Comment:
This validation check may not be required. Because I believe, in the front end the delivery method is sent in one request and the forename and surname is sent in another. I will check the front end designs to see if this is the case.

Reply:
I think the business requirement should be the same - that the name must be provided if the delivery method is collection. This code should enforce that whether the request is the initial create request, or a subsequent update request with any of these fields changed.

Reply:
Cool that's fine. I just checked the design and it asks for the forename and surname first. It asks for the delivery method later on in the journey. https://order-a-cert-prototype-rel1-2.herokuapp.com/order-details